jax/numpy/lax_numpy.py

```diff
@@ -545,6 +545,26 @@ def exp2(x):
   return lax.exp(lax.mul(lax.log(_constant_like(x, 2)), x))
 
 
+@_wraps(onp.signbit)
+def signbit(x):
+  x, = _promote_shapes("signbits", x)
+
+  if issubdtype(_dtype(x), onp.integer):
+    x = lax.convert_element_type(x, onp.float64)
+
+  info = finfo(_dtype(x))
+
+  if info.bits == 16:
+    int_type = onp.int16
+  elif info.bits == 32:
+    int_type = onp.int32
+  elif info.bits == 64:
+    int_type = onp.int64
+
```

Comment:
It would probably be best to defensively raise an assertion error here here just in case none of these conditions match, e.g., float128 is sort of a thing for numpy.

Reply:
Fixed :smiley:

Reply:
`NotImplementedError` would be a better exception to raise.

I'd also reword the message as:
`"jax.numpy.signbit only supports 16, 32, and 64-bit types."`